modalc-lib/modal-vector.rkt

```diff
@@ -0,0 +1,45 @@
+#lang racket
+
+(require modalc)
+
+(define (modal-vector/c should-apply-ctc? . inner-ctc)
```

Comment:
Maybe we should have separate `ref` and `set` modes? And a way to provide one for both. With keywords should work: `#:ref` `#:set` and `#:ref/set` perhaps.